core/bin/oauth.rs

```diff
@@ -203,6 +204,68 @@ async fn connections_access_token(
     }
 }
 
+#[derive(Deserialize)]
+struct CredentialPayload {
+    provider: CredentialProvider,
+    credentials: serde_json::Map<String, serde_json::Value>,
```

Comment:
Let's do serde_json::Value this is the most flexible (any valid JSON)? It is also aligned with what we do for Connections.

Reply:
This name is not ideal (I know I suggested it). Since the top-level object is called a `Credential` we should find something else as this is currently confusing.

content, value, secret ?

Reply:
> Let's do serde_json::Value this is the most flexible (any valid JSON)? It is also aligned with what we do for Connections.

Disagree. If we do this, we get 0 value out of expecting json and we should just expect and store strings. With `serde_json::Value` you validate nothing because any string is a valid json value (i.e a string), so you don't prevent multiple levels of json.stringify.

in my view, it's either `String` (flexible) or `serde_json::Map<String, serde_json::Value>` (safer)

Reply:
Then we want to update metadata in Connections